Dockerfile

```diff
@@ -0,0 +1,72 @@
+# FROM jupyter/base-notebook
+# USER root
+# ENV DEBIAN_FRONTEND=noninteractive
+FROM ubuntu:xenial-20161213
+# Pre-cache neurodebian key
+COPY docker/files/neurodebian-archive-keyring.gpg /root/.neurodebian.gpg
+
+# Prepare environment
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+                    curl \
+                    bzip2 \
+                    ca-certificates \
+                    xvfb \
+                    cython3 \
+                    build-essential \
+                    autoconf \
+                    libtool \
+                    ncurses-dev \
+                    vim \
+                    git \
+                    libmotif-dev \
+                    libnetcdf-dev \
+                    tcsh \
+                    r-base \
+                    python-qt4 \
+                    wget \
+                    pkg-config && \
+    curl -sSL http://neuro.debian.net/lists/xenial.us-ca.full >> /etc/apt/sources.list.d/neurodebian.sources.list && \
+    apt-key add /root/.neurodebian.gpg && \
+    (apt-key adv --refresh-keys --keyserver hkp://ha.pool.sks-keyservers.net 0xA5D32F012649A5A9 || true)
+
+RUN echo deb-src http://neuro.debian.net/debian bionic main contrib> /etc/apt/sources.list.d/neurodebian.list
+
+RUN wget https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/cmake/3.12.1-1/cmake_3.12.1.orig.tar.gz \
+  ; tar xzvf cmake_3.12.1.orig.tar.gz \
+  ; cd cmake-3.12.1 \
+  ; ./bootstrap && make \
+  ; make install \
+  ; cd .. \
+  ;rm -fr cmake* 
+
+# RUN apt-get update & apt-get -y build-dep afni
+RUN apt-get update && \
```

Comment:
the only reason to do `apt-get update` before each `apt-get install` is if at the end you 
```
    && apt-get clean \ 
    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* 
```
to minimize the size of each layer.  Otherwise it is just a waste of time

Reply:
i'll add that now and consolidate the install commands a little now that I'm happier with the dependencies I've specified.